litestar/contrib/pydantic/pydantic_schema_plugin.py

```diff
@@ -322,3 +371,34 @@ def for_pydantic_model(cls, field_definition: FieldDefinition, schema_creator: S
                 )
             ),
         )
+
+    @classmethod
+    def get_required_fields(
+        cls,
+        property_fields: dict[str, FieldDefinition],
+        exclude: PydanticFieldsList = None,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+        exclude_unset: bool = False,
+        include: PydanticFieldsList = None,
+    ) -> list[str]:
+        required = []
+        exclude = exclude or set()
+        include = include or []
+
+        for prop in property_fields.values():
+            name = prop.name
+
+            if any(
+                [
+                    prop.name in exclude,
+                    exclude_none and (prop.is_optional or prop.is_none_type),
+                    (exclude_defaults or exclude_unset) and prop.has_default,
```

Comment:
I have a doubt here. What if the user only wants to ensure that pydantic doesn't include the fields that have a default value (or any of the other cases), when pydantic serializes a model into JSON, but they still want those fields to be documented?

Reply:
I'm still a bit hesitant regarding this. This makes sense for the schema we generate for responses, but it doesn't make sense for the schema we generate for request bodies. Since we don't differentiate between whether the model is used as a request or response body, this may be problematic.

Reply:
I don't think we can since we create the same schema for both request and response bodies. Changing that would be possible, but that could cause issues such as with the naming of the schemas. We would need to name them differently based on whether they were the request or response bodies. I'm not sure if adding that complexity and these changes are worth the benefit.

Reply:
Personally, I think adding it only for `PydanticInitPlugin` would be enough. We can just document that while these flags effect the responses, they're not reflected in the generated OpenAPI schema.

Reply:
> Personally, I think adding it only for `PydanticInitPlugin` would be enough. We can just document that while these flags effect the responses, they're not reflected in the generated OpenAPI schema.

I think that's going to lead to confusion. The core issue here is, and has always been, that you can set conflicting values in various places and it's not immediately clear which ones take precedence. However, I also think that's okay to some degree, as long as we make sure that the defaults and patterns we promote are sensible and the rest of the behaviour is documented well.  

As for this particular problem: If we were to only expose the init plugin for Pydantic, we could ensure internal consistency here, right? So why don't we accept that users will be able to create inconsistent schemas for now if they really want to, and change this in 3.0?